journal/helper.go

```diff
@@ -2,31 +2,48 @@ package journal
 
 import (
 	"strconv"
+	"time"
 
-	"github.com/mitchellh/go-ps"
 	"github.com/puppetlabs/wash/datastore"
+	"github.com/shirou/gopsutil/process"
+	log "github.com/sirupsen/logrus"
 )
 
 // Cache pid to process names. This may end up getting the wrong process name if there are
 // lots of new processes being created constantly, but makes fast things a *lot* faster.
 var pidToExec = datastore.NewMemCache()
 
-// PIDToID converts a process ID to a journal ID, appending the name of the executable
-// running that process if we can retrieve it.
-func PIDToID(pid int) string {
-	pidStr := strconv.FormatInt(int64(pid), 10)
+// PIDToJournalEntry converts a process ID to a journal ID, appending the name of the executable
+// running that process if we can retrieve it - and the command invokation of that process ID.
+func PIDToJournalEntry(pid int) Entry {
```

Comment:
Hmm, could we move this over to https://github.com/puppetlabs/wash/blob/master/journal/core.go#L89, where we trigger this code _if_ the journal file doesn't exist (implying that it's being created for the first time)? We could add it to the history there + maintain a hash to handle the case where the journal file's deleted.

Reply:
The thing we add to context is an opaque blob everywhere else.

I'm not sure what it is you're suggesting be moved.